### PR TITLE
Added Domain Model pattern

### PR DESCRIPTION
**Pull request by AndriyPyzh**
Domain Model pattern

- Resolves issue #1294 
- A Domain Model is an object model of the domain that incorporates both behavior and data.

Description

- Implement domain model pattern 
- corrected main class name in pom.xml of strategy pattern
- minor update of "Related patterns" in readme file of transaction script pattern

### CODE REVIEW

**Comment by tedyoung** [comment]
Domain Model **is not** a Design Pattern, it's a way of modeling and structuring code in complex systems (I'd even say that it's just the right way to do object-oriented programming). Even if it were a pattern, this code directly contradicts the intent by having the "domain objects" directly reference the persistence objects (DAO).

**Comment by iluwatar** [comment]
> Domain Model **is not** a Design Pattern, it's a way of modeling and structuring code in complex systems (I'd even say that it's just the right way to do object-oriented programming). Even if it were a pattern, this code directly contradicts the intent by having the "domain objects" directly reference the persistence objects (DAO).

@tedyoung thanks for the comment! The domain model is listed as one of the domain logic patterns in Fowler's [Patterns of Enterprise Application Architecture](https://www.amazon.com/Patterns-Enterprise-Application-Architecture-Martin/dp/0321127420/ref=sr_1_1?crid=1Y3ZS7XOAUJNU&dchild=1&keywords=patterns+of+enterprise+application+architecture&qid=1625910501&s=books&sprefix=patterns+of+enter%2Caps%2C266&sr=1-1). I definitely think that it's a valid design pattern, but I also understand the confusion around the terms. For example, the book by Evans, [Domain-Driven Design](https://www.amazon.com/Domain-Driven-Design-Tackling-Complexity-Software-dp-0321125215/dp/0321125215/ref=mt_other?_encoding=UTF8&me=&qid=1625910547) is all about designing domain models for applications. The concept is so important and goes so deep into the core of software development that it may feel like something larger than a design pattern.

File: domain-model/src/test/java/com/iluwatar/domainmodel/CustomerDaoImplTest.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+            .build();
+  }
+
+  @AfterEach
+  void tearDown() throws SQLException {
+    TestUtils.deleteSchema(dataSource);
+  }
+
+  @Test
+  void findByNameTest() throws SQLException {
```
Comment (iluwatar) [inline domain-model/src/test/java/com/iluwatar/domainmodel/CustomerDaoImplTest.java:79]: We don't need to have the "test" suffix in the test method names. The test method name should always describe what is being tested.

Comment (iluwatar) [reply domain-model/src/test/java/com/iluwatar/domainmodel/CustomerDaoImplTest.java:79]: See some examples here https://stackoverflow.com/questions/155436/unit-test-naming-best-practices

**Review by iluwatar — CHANGES_REQUESTED** [review]

**Comment by AndriyPyzh** [comment]
@iluwatar thanks for the review! I updated PR according to your suggestions.

File: domain-model/README.md [hunk 2]
```diff
@@ -0,0 +1,314 @@
+---
+layout: pattern
+title: Domain Model
+folder: domain-model
+permalink: /patterns/domain-model/
+categories: Structural
+tags:
+ - Data access
```
Comment (iluwatar) [inline domain-model/README.md:8]:
```suggestion
 - Domain
```

File: domain-model/README.md [hunk 3]
```diff
@@ -0,0 +1,314 @@
+---
+layout: pattern
+title: Domain Model
+folder: domain-model
+permalink: /patterns/domain-model/
+categories: Structural
```
Comment (iluwatar) [inline domain-model/README.md:6]:
```suggestion
categories: Architectural
```

**Review by iluwatar — CHANGES_REQUESTED** [review]
I think it looks good already, but I'd like to change the category and tags for the pattern.

**Comment by AndriyPyzh** [comment]
@iluwatar I committed your changes.

**Review by iluwatar — APPROVED** [review]

**Comment by iluwatar** [comment]
Thank you for the new pattern!
@all-contributors please add @AndriyPyzh for code

**Pull request merged**